composer/tools/composer_db_transfer.md

```diff
@@ -0,0 +1,160 @@
+    script requires permissions to access the Composer environment
+    (roles/composer.environmentAndStorageObjectAdmin role is sufficient) and its
+    GKE cluster (container.*).
+
+1.  Make sure that you can access GKE cluster of Composer environment through
+    kubectl (e.g. if you are not able to successfully execute
+    `kubectl get pods`, the script will fail). "Unable to connect to the server:
+    dial tcp [IP ADDRESS]:443: connect: connection timed out" reported by
+    kubectl indicates that kubectl can not connect to GKE cluster. Check
+    https://cloud.google.com/kubernetes-engine/docs/how-to/private-clusters#cloud_shell
+    for instructions about setting up access to private GKE cluster if you have
+    private IP environment.
+
+1.  Supported source versions: all Composer versions with Airflow 1.10.14 or
+    1.10.15
+
+1.  Supported destination versions: all Composer versions with Airflow 2.0.1 and
+    newer
+
+1.  The script depends on Python 3.6 (or newer), gcloud and kubectl. Make sure
+    you have all those tools installed.
+
+1.  Make sure that your Composer environment is healthy ("Environment health",
+    "Database health" - see:
+    https://cloud.google.com/composer/docs/monitoring-dashboard). If it is not
+    healthy, first step is to fix the environment.
+
+1.  Make sure that you have write access to the current directory. The script
+    exports KUBECONFIG environment variable to store credentials to the
+    Composer's GKE cluster in current directory (on the machine that executes
+    the script). The purpose is not to interfere with existing GKE-related
+    user's configuration.
+
+1.  Pause the DAGs before exporting the database.
+
+## Limitations
+
+1.  Logs from old DAG executions are not transferred (but history of executions
+    in the database is transferred).
+
+1.  The script does not perform any DAG modifications, does not validate that
+    the DAGs are suitable to run in Airflow 2.0 and does not verify that
+    suitable dependencies are installed in target environment. Airflow
+    [upgrade_check](https://airflow.apache.org/docs/apache-airflow/stable/upgrade-check.html)
+    might be used to troubleshoot DAG compatibility issues.
+
+1.  `dag_code` and `serialized_dag` tables are not synchronized directly.
+    Database entries for imported DAGs are created by Airflow in the new
+    environment.
+
+1.  The script does not prevent DAGs from being scheduled in both old and new
+    environment. If can be prevented e.g. by pausing the DAGs before exporting
+    them and unpausing them in the new environment.
+
+1.  Airflow web server in target environment might temporarily not be available
+    after the database transfer.
+
+1.  The environment might need some time to parse imported DAGs. As a result
+    they might appear in the web UI with some delay.
+
+1.  Import operation removes all records from the target environment. Files
+    existing in environment's bucket are not deleted.
```

Comment:
```suggestion
1.  Import operation removes all records from the target environment. Files
    existing in the environment's bucket are not deleted.
```

Reply:
Each operation is referring to only one environment specified in parameters.